slurm_openstack_tools/sacct.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+import datetime
+import json
+import subprocess
+import re
+
+from ClusterShell import NodeSet
+
+SLURM_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S"
+TIMESTAMP_FILE = "lasttimestamp"
+
+def main():
+    args = ["sacct", "-X", "--allusers", "--parsable2", "--format",
+        "jobid,jobidraw,cluster,partition,account,group,gid,"
+        "user,uid,submit,eligible,start,end,elapsed,elapsedraw,exitcode,state,nnodes,"
+        "ncpus,reqcpus,reqmem,reqgres,reqtres,timelimit,nodelist,jobname",
+        "--state",
+        "CANCELLED,COMPLETED,FAILED,NODE_FAIL,PREEMPTED,TIMEOUT"]
+    # Work out starttime and endtime
+    now = datetime.datetime.utcnow()
+    end_str = now.strftime(SLURM_DATE_FORMAT)
+
+    try:
+        with open(TIMESTAMP_FILE) as f:
+            start_str = f.read()
+    except FileNotFoundError:
+        # Default to start of today
+        start_str = "00:00:00"
+
+    args += ["--starttime", start_str]
```

Comment:
I think here we need:

if start_str:
  args += ["--starttime", start_str]

Reply:
Went for collecting jobs from up to a year ago. Using a date too far in the past resulted in no output :-/